### PR TITLE
CCR: Requires soft-deletes on the follower

### PR DESCRIPTION
**Pull request by dnhatn**
Since #34412 and #34474, a follower must have soft-deletes enabled to
work correctly. This change requires soft-deletes on the follower.

Relates #34412
Relates #34474

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/index/engine/FollowingEngine.java [hunk 1]
```diff
@@ -49,6 +49,9 @@ private static EngineConfig validateEngineConfig(final EngineConfig engineConfig
         if (CcrSettings.CCR_FOLLOWING_INDEX_SETTING.get(engineConfig.getIndexSettings().getSettings()) == false) {
             throw new IllegalArgumentException("a following engine can not be constructed for a non-following index");
         }
+        if (engineConfig.getIndexSettings().isSoftDeleteEnabled() == false) {
+            throw new IllegalArgumentException("a following engine requires soft-deletes enabled");
```
Comment (jasontedor) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/index/engine/FollowingEngine.java]: Nit: `s/soft-deletes enabled/soft deletes to be enabled`

**Review by jasontedor — APPROVED** [review]
LGTM. I left one comment.

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
Thanks @jasontedor and @martijnvg.

**Pull request merged**